rebuild-awstats/rebuild-awstats.sh

```diff
@@ -223,15 +223,12 @@ merge_logs() {
     $AWSTATS_TOOLS_D/logresolvemerge.pl "$@" 
 }
 
-check_log_sort(){
+check_log_sort() {
     local log_file=$1
 
-    LC_TIME=C sort -s -t ' ' -k 4.9n -k 4.5M -k 4.2n -C "$log_file" && return
-
-    echo "WARNING: Сontents of log files not sorted, will running sorting logs, this operation may take a long time"
-
-    LC_TIME=C sort -S 1% -s -t ' ' -k 4.9n -k 4.5M -k 4.2n -o "$log_file.sort" "$log_file"
-    mv -f "$log_file.sort" "$log_file"
+    ! LC_TIME=C sort -s -t ' ' -k 4.9n -k 4.5M -k 4.2n -k 4.14 -C "$log_file" || return 0
+    echo "WARNING: Log files are not sorted by timestamp. They will be sorted, but this operation is resource-intensive and may take a long time."
+    ! LC_TIME=C sort -S 1% -s -t ' ' -k 4.9n -k 4.5M -k 4.2n -k 4.14 -o "$log_file.sort" "$log_file" || mv -f "$log_file.sort" "$log_file"
```

Comment:
In this case I would suggest something like `sort ... && mv ... || rm -f "$log_file.sort"` to make sure extra data copies aren't left around in case of some failures.